java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -389,10 +413,22 @@ public void pushBlocking() throws IOException {
         Preconditions.checkNotNull(client, "client must not be null.");
         Preconditions.checkArgument(client.isInitialize(), "client must be initialized.");
         Preconditions.checkArgument(client.getSyncManager().getCount(getCollectionName()) == 0, "InvalidOperation. You must push all pending sync items before new data is pulled. Call push() on the data store instance to push pending items, or purge() to remove them.");
-
-        KinveyAbstractReadResponse<T> response = new KinveyAbstractReadResponse<T>();
         query = query == null ? client.query() : query;
+        if (deltaSetCachingEnabled && query.getSkip() == 0 && query.getLimit() == 0) {
+            return getBlockingDeltaSync(query);
+        } else {
+            return pullBlockingNoDeltaSync(query);
+        }
```

Comment:
Why does one method name start with `get`, and the other with `pull`?

Reply:
Updated.